test/test_cuda.py

```diff
@@ -4659,9 +4441,212 @@ def test_no_triton_on_import(self):
         self.assertEqual(rc, "False", "Triton was imported when importing torch!")
 
 
+@torch.testing._internal.common_utils.markDynamoStrictTest
 class TestCudaOptims(TestCase):
     # These tests will be instantiate with instantiate_device_type_tests
     # to apply the new OptimizerInfo structure.
+    def _test_graphed_optims(self, steps_warmup, steps_train, optimizer_ctor, kwargs):
+        for actually_do_graphs in (True, False):
+            params = [torch.randn((i + 5, i + 5), device="cuda") for i in range(2)] + [
```

Comment:
```suggestion
            params = [torch.randn((i + 5, i + 5), device=device) for i in range(2)] + [
```
all devices should use the `device` arg, and same with `dtype`